### PR TITLE
Feat: add more logs to code and log validators request in ejector module

### PR DESCRIPTION
**Pull request by F4ever**

### CODE REVIEW

File: src/modules/ejector/ejector.py [hunk 1]
```diff
@@ -105,6 +106,19 @@ def build_report(self, blockstamp: ReferenceBlockStamp) -> tuple:
...
+                f.write(
+                    json.dumps([{
+                        **v.__dict__,
+                        'validator': v.validator.__dict__,
+                    } for v in self.w3.cc.get_validators(blockstamp)])
+                )
+        except Exception as error:  # pylint: disable=W0703
+            logger.error({'msg': 'Failed to save get_validators response.', 'error': str(error)})
+        else:
+            logger.info({'msg': 'Response get_validators from Consensus Cline wrote to file validators_response.json.'})
```
Comment (skhomuti) [inline src/modules/ejector/ejector.py:120]:
```suggestion
            logger.info({'msg': 'Response get_validators from Consensus Client written to validators_response.json.'})
```

File: src/modules/ejector/ejector.py [hunk 2]
```diff
@@ -105,6 +106,19 @@ def build_report(self, blockstamp: ReferenceBlockStamp) -> tuple:
 
         EJECTOR_VALIDATORS_COUNT_TO_EJECT.set(report_data.requests_count)
 
+        try:
+            with open('validators_response.json', 'w') as f:
+                f.write(
+                    json.dumps([{
+                        **v.__dict__,
+                        'validator': v.validator.__dict__,
```
Comment (skhomuti) [inline src/modules/ejector/ejector.py:114]: it works like a charm including nested dataclasses too
```
from dataclasses import asdict 

asdict(v)
```

File: src/modules/ejector/ejector.py [hunk 3]
```diff
@@ -261,8 +275,7 @@ def _get_predicted_withdrawable_epoch(
 
         churn_limit = self._get_churn_limit(blockstamp)
 
-        remain_exits_capacity_for_epoch = churn_limit - latest_to_exit_validators_count
-        epochs_required_to_exit_validators = (validators_to_eject_count - remain_exits_capacity_for_epoch) // churn_limit + 1
+        epochs_required_to_exit_validators = (validators_to_eject_count + latest_to_exit_validators_count) // churn_limit
```
Comment (skhomuti) [inline src/modules/ejector/ejector.py:278]: still not sure whether it really needs to change

**Review by skhomuti — CHANGES_REQUESTED** [review]

File: src/modules/ejector/ejector.py (hunk 3 above)
Comment (F4ever) [reply src/modules/ejector/ejector.py:278]: The result is the same, I just simplified the expression

Comment (F4ever) [reply src/modules/ejector/ejector.py:278]: you can do it on paper and check it

File: src/modules/ejector/ejector.py (hunk 2 above)
Comment (F4ever) [reply src/modules/ejector/ejector.py:114]: lol, it's very slow.
My method works 0.6107 seconds
asdict works 9.8773 seconds
with same input data (500 MB of validators)

**Review by skhomuti — DISMISSED** [review]

**Review by vgorkavenko — DISMISSED** [review]

File: src/modules/ejector/ejector.py (hunk 3 above)
Comment (infloop) [reply src/modules/ejector/ejector.py:278]: I'm here against any changes in logic, even simplification of expressions

**Review by vgorkavenko — APPROVED** [review]

**Review by infloop — APPROVED** [review]

**Pull request merged**